app/src/main/java/com/github/kulebin/myfoursquareapp/imageLoader/ImageLoader.java

```diff
@@ -207,4 +233,14 @@ private void setPlaceholder(final ImageView pView, final int pResId) {
             pView.setImageResource(android.R.color.transparent);
         }
     }
+
+    private void addBitmapToCache(final String pUrl, final Bitmap pBitmap) {
+        if (getBitmapFromCache(pUrl) == null) {
```

Comment:
If you reach this method - you will have no image in cache, so it's better just to put twice image to cache in rare edge cases, then to check if image is already in cache every time.

Optimization tip: do logic that is faster for biggest part of user first